java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java

```diff
@@ -282,8 +282,8 @@ else if (ksession != null) {
 
         if (map.containsKey(CHILD) && !Config.get().getBoolean("ks_restrict_child_channels")) {
             Channel child = ChannelFactory.lookupByLabel(map.get(CHILD));
-            if (child == null || tree == null || child.getParentChannel() == null ||
-                    !child.getParentChannel().equals(tree.getChannel())) {
+            if (child == null || tree == null || (child.getParentChannel() != null &&
+                    !child.getParentChannel().equals(tree.getChannel()))) {
```

Comment:
this was changed for fixing https://github.com/uyuni-project/uyuni/commit/7713baf8771d79cfd50c3bcdb3aa7fe866560ad3. Are you sure `child.getParentChannel() == null` is no longer required? What I meant is having:
```
            if (child == null || tree == null || child.getParentChannel() == null || (child.getParentChannel() != null &&
                    !child.getParentChannel().equals(tree.getChannel()))) {
```